xarray/core/formatting.py

```diff
@@ -253,8 +253,15 @@ def summarize_coord(name, var, col_width):
 
 
 def summarize_attr(key, value, col_width=None):
-    # ignore col_width for now to more clearly distinguish attributes
-    return u'    %s: %s' % (key, maybe_truncate(value))
+    """Summary for __repr__ - use ``X.attrs[key]`` for full value."""
+    key = u'    %s:' % unicode_type(key)
+    if col_width is not None:
+        key = pretty_print(key, col_width)
+    output = u'%s %s' % (key, unicode_type(value).replace(u'\t', u'    '))
```

Comment:
If somebody is putting non-strings in attributes (which they almost certainly are, as ill-advised as that may be), there is some value in having them appear in a noticeably distinct way. I would rather have non-strings look a little funky and span multiple lines than be confused with the `repr` of the objects.

e.g., consider how the repr of an `xarray.Dataset` looks:
```
In [16]: repr(xarray.Dataset({'foo': ('x', [1, 2, 3])}))[:70] + '...'
Out[16]: '<xarray.Dataset>\nDimensions:  (x: 3)\nDimensions without coordinates: x...'
```

Though, it might possibly make sense to "indent" their reprs on any newlines to align them with the indentation of first line.

@fmaussion @pwolfram any opinions here?